javaslang/src/main/java/javaslang/collection/HashMap.java

```diff
@@ -821,20 +821,13 @@ public int size() {
     }
 
     @Override
-    public int hashCode() {
-        return trie.hashCode();
+    public boolean equals(Object o) {
+        return Collections.equals(this, o);
     }
 
     @Override
-    public boolean equals(Object o) {
-        if (o == this) {
-            return true;
-        } else if (o instanceof HashMap) {
-            final HashMap<?, ?> that = (HashMap<?, ?>) o;
-            return this.trie.equals(that.trie);
-        } else {
-            return false;
-        }
+    public int hashCode() {
+        return Collections.hash(this);
```

Comment:
I believe this is wrong way. `HAMT`'s implementations of `equals()` and `hashCode()` are more complex than usual (see `HAMTTest::shouldEqualsIgnoreOrder()` and #1823 #1815). This behavior is not checked in the `HashMap` and this is a bug in the tests